test/units/testsuite-04.sh

```diff
@@ -5,6 +5,9 @@ set -o pipefail
 
 # Test stdout stream
 
+# https://github.com/systemd/systemd/issues/15528
+journalctl --follow --file=/var/log/journal/*/* | read -n1
```

Comment:
*sigh*, thanks to the interactivity of `--follow` this always returns 1 with `set -o pipefail` as we *interrupt* it:

```
$ set -o pipefail
$ journalctl --follow | head -n1
-- Logs begin at Mon 2019-09-30 22:08:02 CEST. --
$ echo $?
1
```

One possible solution could be this:
```diff
$ git diff
diff --git a/test/units/testsuite-04.sh b/test/units/testsuite-04.sh
index f6cc3dee29..c82cee2a1b 100755
--- a/test/units/testsuite-04.sh
+++ b/test/units/testsuite-04.sh
@@ -6,7 +6,7 @@ set -o pipefail
 # Test stdout stream
 
 # https://github.com/systemd/systemd/issues/15528
-journalctl --follow --file=/var/log/journal/*/* | read -n1
+journalctl --follow --file=/var/log/journal/*/* | head -n1 || [[ $? -eq 1 ]]
 
 # Skip empty lines
 ID=$(journalctl --new-id128 | sed -n 2p)
```

Which will handle the non-zero EC code for "standard" stream interrupt, but will fail for exit codes:

```
## Fixed version
$ journalctl --follow --file=/var/log/journal/*/* | head -n1 || [[ $? -eq 1 ]]
-- Logs begin at Fri 2019-09-06 22:29:12 CEST. --
$ echo $?
0

## Borked version
$ journalctl --follow --file=/var/log/journal/*/* | head -n1 || [[ $? -eq 1 ]]
Segmentation fault (core dumped)
$ echo $?
1
```

Also, `head` is used instead of `read`, since `read`, for some reason, masks the SIGSEGV signal.